server.js

```diff
@@ -0,0 +1,115 @@
+const express = require('express');
+const mongoose = require('mongoose');
+const session = require('express-session');
+const server = express();
+const User = require('./authenticate/userModel');
+
+mongoose.connect('mongodb://localhost/auth-i').then(() => {
+    console.log('\n *** ALL SYSTEMS ARE A GO!!! ***\n');
+});
+
+//middleware
+const sessionSetup = {
+    secret: "I am the very model of a modern major general!",
+    cookie : {
+        maxAge: 1000 * 60 * 60 * 24 //1 day in MS
+    },
+    httpOnly: true,
+    secure: false,
+    resave: true,
+    saveUninitialized: true,
+    name: "you wish",
+};
+
+function onlyLoggedIn(req, res, next) {
+    if (req.session && req.session.username) {
+        next();
+    } else {
+        res.status(401).send("NOT MY SERVER B***H!!!!")
+    }
+}
+
+server.use(express.json());
+server.use(session(sessionSetup));
+
+server.get('/api/restricted/users', onlyLoggedIn, (req, res) => {
+    User
+        .find()
+        .then(users => res.json(users))
+        .catch(err => res.json(err));
+})
+
+server.get('/', (req, res) => {
+    if (req.session && req.session.username) {
+        res.status(200).json({message: `Welcome Back ${req.session.username}`});
+    } else {
+        res.status(400).json({message: "IDENTIFY YOURSELF!!!"});
+    }
+    
+});
+
+server.post('/api/register', (req, res) => {
+    User
+        .create(req.body)
+        .then(user => {
+            res.status(201).json(user);
+        })
+
+        .catch(error => {
+            if(error.code === 11000) {
+                res.status(500).json({error: "Username already taken!"})
```

Comment:
I like that you went on the journey to identify the MongoDB errors for duplicate unique fields and implemented it into the error checking portion of the registration.